src/filter/kuadrant_filter.rs

```diff
@@ -600,6 +631,271 @@ impl KuadrantFilter {
             response_headers_to_add: Some(Vec::default()),
             request_headers_to_add: Some(Vec::default()),
             phase: Phase::RequestHeaders,
+            response_content_type: None,
+            sse_buffer: String::new(),
+            sse_usage: String::new(),
+            sse_offset: 0,
+        }
+    }
+
+    fn handle_sse_stream(
```

Comment:
Maybe not for this PR. But I would have implemented this `handle_sse_stream` in a generic way, expecting generic Server Sent Events. This method would produce events and some other function may use those events as source for finding `usage` data.